File: src/main/java/com/indiepost/model/analytics/Click.java
package com.indiepost.model.analytics;

import javax.persistence.*;

/**
 * Created by jake on 8/7/17.
 */
@Entity
@DiscriminatorValue("Click")
public class Click extends Stat {

    @ManyToOne
    @JoinColumn(name = "linkId")
    private Link link;

    @Column(updatable = false, insertable = false)
    private Long linkId;

    public Long getLinkId() {
        return linkId;
    }

    public void setLinkId(Long linkId) {
        this.linkId = linkId;
    }

    public Link getLink() {
        return link;
    }

    public void setLink(Link link) {
        this.link = link;
    }
}

File: src/main/kotlin/com/indiepost/model/analytics/Click.kt
package com.indiepost.model.analytics

import javax.persistence.*

/**
 * Created by jake on 8/7/17.
 */
@Entity
@DiscriminatorValue("Click")
class Click : Stat() {
    @ManyToOne
    @JoinColumn(name = "linkId")
    var link: Link? = null

    @Column(updatable = false, insertable = false)
    var linkId: Long? = null
}
